fdbserver/Resolver.actor.cpp

```diff
@@ -416,7 +415,7 @@ ACTOR Future<Void> resolveBatch(Reference<Resolver> self, ResolveTransactionBatc
 				reply.tpcvMap.clear();
 			} else {
 				std::set<uint16_t> writtenTLogs;
-				if (shardChanged || reply.privateMutationCount) {
+				if (shardChanged || toCommit->getAllMessages().size() > 1) {
```

Comment:
Sorry, missed this part: is it "size() > 0" or "size() > 1"?

Reply:
This is not right. `getAllMessages()` returns all locations' messages, including empty ones, so it's not the one we are looking for.

So I think it should be `if (shardChanged || reply.privateMutationCount > 1)`, since `reply.privateMutationCount = toCommit->getMutationCount();`, which just returns the subsequence (starting at 1). Alternatively, we can fix `getMutationCount()` to return `subsequence - 1`, which is better.